File: Room.cpp
#include "Room.hpp"
#include <iostream>

Room::Room(const std::string& name) : m_name(name) {}

Room::~Room()
{
	// Удаление всех динамически созданных датчиков.
	for (auto s : m_sensors)
		delete s;
}

void Room::AddSensor(Sensor* sensor)
{
	m_sensors.push_back(sensor);
}

void Room::CheckAllSensors()
{
	std::cout << "Checking sensors in room: " << m_name << std::endl;
	for (auto s : m_sensors)
	{
		s->ReadValue();
		s->CheckStatus();
	}
}

std::string Room::GetName() const
{
	return m_name;
}

File: Lab04/Lab04Monitoring/Room.cs
using System;
using System.Collections.Generic;
using System.Globalization;

namespace Lab04Monitoring
{
    // Класс Room описывает помещение, в котором установлены датчики.
    public class Room
    {
        // Имя комнаты.
        public string Name { get; }

        // Список датчиков в комнате.
        private readonly List<Sensor> _sensors = new();

        public Room(string name)
        {
            Name = name;
        }

        // Добавление датчика.
        public void AddSensor(Sensor sensor)
        {
            _sensors.Add(sensor);
        }

        // Опрос всех датчиков и вывод результатов.
        public void CheckAllSensors()
        {
            Logger.Instance.Log($"Checking sensors in room '{Name}'");

            foreach (var s in _sensors)
            {
                var value = s.ReadValue();
                var ok = s.CheckStatus();

                Console.WriteLine(
                    $"Room '{Name}': {s.Name} = {value.ToString("F2", CultureInfo.InvariantCulture)} " +
                    $"{s.Unit}, status={s.Status}, ok={ok}"
                );
            }
        }
    }
}
